modules/segmentshortestpaths/core/segmmetricshortestpath.cpp

```diff
@@ -0,0 +1,159 @@
+// sala - a component of the depthmapX - spatial network analysis platform
+// Copyright (C) 2000-2010, University College London, Alasdair Turner
+// Copyright (C) 2011-2012, Tasos Varoudis
+// Copyright (C) 2017-2018, Petros Koutsolampros
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "segmmetricshortestpath.h"
+
+#include "genlib/stringutils.h"
+
+bool SegmentMetricShortestPath::run(Communicator *) {
+
+    AttributeTable &attributes = m_map.getAttributeTable();
+
+    bool retvar = true;
+
+    int dist_col = attributes.insertOrResetColumn("Metric Shortest Path Distance");
+    int path_col = attributes.insertOrResetColumn("Metric Shortest Path Order");
+
+    // record axial line refs for topological analysis
+    std::vector<int> axialrefs;
+    // quick through to find the longest seg length
+    std::vector<float> seglengths;
+    float maxseglength = 0.0f;
+    for (size_t cursor = 0; cursor < m_map.getShapeCount(); cursor++) {
+        AttributeRow &row = m_map.getAttributeRowFromShapeIndex(cursor);
+        axialrefs.push_back(row.getValue("Axial Line Ref"));
+        seglengths.push_back(row.getValue("Segment Length"));
+        if (seglengths.back() > maxseglength) {
+            maxseglength = seglengths.back();
+        }
+    }
+
+    int maxbin = 512;
+
+    std::vector<unsigned int> seen(m_map.getShapeCount());
+    std::vector<TopoMetSegmentRef> audittrail(m_map.getShapeCount());
+    std::vector<int> list[512]; // 512 bins!
```

Comment:
I have no idea what you are doing with 512 bins! Could you either explain the algo in a comment (or, maybe better, reference a paper explaining it)?